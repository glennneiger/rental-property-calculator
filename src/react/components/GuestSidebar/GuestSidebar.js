import React, { Component } from 'react';
import PropTypes from 'prop-types';

import css from './guestSidebar.css';
import BlueButton from '../BlueButton';

class GuestSidebar extends Component {
  onLoginClick = () => {
    this.props.history.push('/login');
  }
  onRegisterClick = () => {
    this.props.history.push('/register');
  }
  render() {
    return (
      <div className={css.guestSidebar}>
        <BlueButton onClick={this.onLoginClick}>Login</BlueButton>
        <p>or</p>
        <BlueButton onClick={this.onRegisterClick}>Register</BlueButton>
        <p>to save your analysis</p>
      </div>
    );
  }
}

GuestSidebar.propTypes = {
  history: PropTypes.object
};

export default GuestSidebar;