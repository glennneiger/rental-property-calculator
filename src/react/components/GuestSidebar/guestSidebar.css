.guestSidebar {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.guestSidebar p {
  font-size: 24px;
  text-align: center;
}