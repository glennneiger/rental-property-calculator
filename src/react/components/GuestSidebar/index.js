import { withRouter } from 'react-router-dom';

import GuestSidebar from './GuestSidebar';

export default withRouter(GuestSidebar);