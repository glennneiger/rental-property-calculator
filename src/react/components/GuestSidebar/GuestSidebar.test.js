import React from 'react';
import { shallow } from 'enzyme';
import { expect as chaiExpect } from 'chai';

import GuestSidebar from './GuestSidebar';

describe('<GuestSidebar />', () => {
  let wrapper;

  beforeAll(() => {
    wrapper = shallow(<GuestSidebar />);
  });

  test('container has proper className for styling', () => {
    chaiExpect(wrapper).to.have.className('guestSidebar');
  });

  it('renders 2 BlueButtons', () => {
    chaiExpect(wrapper.find('BlueButton')).to.have.length(2);
  });

  it('renders 2 paragraphs', () => {
    chaiExpect(wrapper.find('p')).to.have.length(2);
  });

  it('has the Login button as the first BlueButton', () => {
    chaiExpect(wrapper.find('BlueButton').at(0).prop('children'))
      .to.equal('Login');
  });

  it('has the Register button as the second BlueButton', () => {
    chaiExpect(wrapper.find('BlueButton').at(1).prop('children'))
      .to.equal('Register');
  });

  it('calls correct function when Login BlueButton is clicked', () => {
    const onLoginClickSpy = jest.spyOn(GuestSidebar.prototype, 'onLoginClick')
      .mockImplementation();

    const component = shallow(<GuestSidebar />);

    component.find('BlueButton').at(0).simulate('click');

    expect(onLoginClickSpy).toHaveBeenCalledTimes(1);

    onLoginClickSpy.mockRestore();
  });
});