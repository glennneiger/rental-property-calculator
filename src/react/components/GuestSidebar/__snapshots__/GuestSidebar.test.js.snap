// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GuestSidebar /> renders correctly 1`] = `
<div
  className="guestSidebar"
>
  <button
    className="blueButton"
    onClick={[Function]}
  >
    Login
  </button>
  <p>
    or
  </p>
  <button
    className="blueButton"
    onClick={[Function]}
  >
    Register
  </button>
  <p>
    to save your analysis
  </p>
</div>
`;
